### DISCUSSION TITLE
SDL3 on Ubuntu

### DISCUSSION THREAD

**Discussion by isolin** [amerkoleci/Vortice.Vulkan#49] [Q&A] [answered]
Hi! I would like to create a specialized off-screen HW raytracer and Vortice.Vulkan could be a good basis for it. I tried to run the triangle sample, but failed at the point when SDL3 is triggered for the first time. It seems that it is missing in my system, but I haven't found a way to install it. I am on Ubuntu 20.04. On of the previous questions mentions the [native libs repo](https://github.com/amerkoleci/native_libs), but in case I compile it, I am not sure how to bind it. A bit of help would be appreciated.

I would also like to ask if SDL3 is the only way of getting a `vkDevice`? Since I will need no rendering window at the end, just a framebuffer, is there maybe a lighter alternative to SDL?

**Comment by amerkoleci** [comment]
Hi,
Not sure I understand what the issue is, SDL native libs are bundled within the nuget package. 

What error do you get?

**Reply by isolin** [reply, marked as answer]
Thank you for the quick reply!

```
Unhandled exception. System.DllNotFoundException: Unable to load shared library 'SDL3' or one of its dependencies. In order to help diagnose loading problems, consider using a tool like strace. If you're using glibc, consider setting the LD_DEBUG environment variable: 

#it follows a list of location proposals, none of them matches - either wrong path or wrong filename
```

In the meanwhile, found `libSDL3.so in `bin/Debug/net8.0/runtimes/linux-x64/native`, but apparently it does not look for the exact combination of path and filename, hence the error.

UPDATE: It also complains that
```
/lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /home/ubuntu/Code/VorticeVulkanTest/bin/Debug/net8.0/runtimes/linux-x64/native/libSDL3.so)
```
so maybe that is the issue, that it finds `libSDL3.so` but **it was compiled against a higher version of GLIBC**.

**Reply by amerkoleci** [reply]
Yeah, could be, try to install the required GLIBC and see if it works!

**Reply by isolin** [reply]
I have setup a new container with Ubuntu 24.04 and it runs fine now. Maybe you can add the minimum GLIBC version as a note to the readme. 20.04 for sure has a lower version, not sure about 22.04.
